src/agent/manager.cpp

```diff
@@ -860,22 +864,41 @@ Anonymous* createOverlayAgentManager(const Parameters& parameters)
 
   if (agentConfig.isNone()) {
     LOG(ERROR) << "Missing `agent_config`";
+
     return nullptr;
   }
 
+  Option<string> master = os::getenv(MESOS_MASTER);
+  if (master.isNone() && agentConfig->has_master()) {
+    master = agentConfig->master();
```

Comment:
Are you sure this is the logic you want? EnvVar trumps explicit module config? Not vice versa? (Just clarifying)